### PR TITLE
Extract exclude rules to separate files, abstract away from Gradle.

### LINKED ISSUE #19: Extract local and remote excludes as configurable parameters
lines like 
```--exclude='artifacts' \```
or 
```--exclude='kotlin' \``` 
show that it is still bound to a particular project. 

It might me a good idea to extract it. 

As separate files?
```.localignore```
```.remoteignore```
?

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #19.

This allows `mainframer.sh` to support basically *any* build system, see `samples/gradle` for gradle example.

### CODE REVIEW

File: mainframer.sh [hunk 1]
```diff
@@ -40,32 +42,39 @@ if [ -z "$BUILD_COMMAND" ]; then
 	exit 1
 fi
 
-pushd "$PROJECT_DIR"
+# Read local exclude rules.
+LOCAL_EXCLUDE_FILE=".mainframerignorelocal"
```
Comment (dmitry-novikov) [inline mainframer.sh]: What if we name this files `.mfignore.local` and `.mfignore.remote`?

Comment (artem-zinnatullin) [inline mainframer.sh]: I'm thinking about separate folder for mainframer configs where we could place .ignorelocal,  .ignoreremote and #50 .config

File: mainframer.sh [hunk 2]
```diff
@@ -40,32 +61,21 @@ if [ -z "$BUILD_COMMAND" ]; then
 	exit 1
 fi
 
-pushd "$PROJECT_DIR"
+# Read exclude rules.
+LOCAL_EXCLUDE="$(readExcludesFile .mainframerignorelocal)"
```
Comment (ming13) [inline mainframer.sh]: Not sure about that. What about `.mainframerlocalignore`?

File: mainframer.sh [hunk 3]
```diff
@@ -40,32 +61,21 @@ if [ -z "$BUILD_COMMAND" ]; then
...
-rsync --archive --delete --compress-level=$LOCAL_COMPRESS_LEVEL \
---exclude='.gradle' \
---exclude='.idea' \
---exclude='**/.git/' \
---exclude='artifacts' \
---exclude='captures' \
---exclude='**/build' \
---exclude='**/local.properties' \
---rsh "ssh" ./ "$REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME"
+eval "rsync --archive --delete --compress-level=$LOCAL_COMPRESS_LEVEL $LOCAL_EXCLUDE --rsh ssh ./ $REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME"
```
Comment (ming13) [inline mainframer.sh]: At this point we can make this a function with four arguments. Not sure if it’s worth it.

File: samples/gradle/.mainframerignoreremote [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.gradle
```
Comment (ming13) [inline samples/gradle/.mainframerignoreremote]: Samples don’t entirely repeat what we have at this point.

File: mainframer.sh [hunk 5]
```diff
@@ -40,32 +61,21 @@ if [ -z "$BUILD_COMMAND" ]; then
...
+
 # Sync project to remote machine.
-rsync --archive --delete --compress-level=$LOCAL_COMPRESS_LEVEL \
---exclude='.gradle' \
---exclude='.idea' \
---exclude='**/.git/' \
---exclude='artifacts' \
---exclude='captures' \
---exclude='**/build' \
---exclude='**/local.properties' \
```
Comment (ming13) [inline mainframer.sh]: Remote and local excludes look very similar. `.mainframerignorecommon`?

Comment (artem-zinnatullin) [inline mainframer.sh]: No, that's totally up to each project members

File: mainframer.sh [hunk 6]
```diff
@@ -15,6 +15,27 @@ function property {
     grep "^${1}=" $PROJECT_DIR/local.properties | cut -d'=' -f2
 }
 
+function readExcludesFile {
```
Comment (artem-zinnatullin) [inline mainframer.sh]: Nothing, will try

File: samples/gradle/.mainframerignoreremote (hunk 4 above)
Comment (artem-zinnatullin) [inline samples/gradle/.mainframerignoreremote]: Yup, it's basic `gradle` sample, not `gradle-android`

Comment (ming13) [inline samples/gradle/.mainframerignoreremote]: `gradle-android` will make more sense since our userbase mostly consists of Android developers (for now).

Comment (artem-zinnatullin) [inline samples/gradle/.mainframerignoreremote]: Will do in a separate pr.

File: mainframer.sh (hunk 2 above)
Comment (artem-zinnatullin) [inline mainframer.sh]: See #54. I'd vote for current names at the moment and then if we decide to move configs into a separate folder -> rename this files, names are not the point of this PR.

File: mainframer.sh (hunk 5 above)
Comment (artem-zinnatullin) [inline mainframer.sh]: Though, ok, this looks like a valid feature, let's think about it and add later if we'll agree on it

**Comment by artem-zinnatullin** [comment]
Switched to rsync builtin `--exclude-from`, PTAL!

I'd vote for merging as is and adding other things in separate PRs.

File: mainframer.sh (hunk 3 above)
Comment (artem-zinnatullin) [inline mainframer.sh]: No need for now
